R/tasas_sf.R

```diff
@@ -0,0 +1,230 @@
+
+#' Download interest rates for savings
+#'
+#' Get the series of the interest rates for savings in the Dominican Republic
+#' separated by time and type of saving
+#'
+#' @return a data frame with monthly series
+#' @export
+get_tasas_pasivas <- function() {
+  file_names <- c(
+    "Tasas 1991-2007",
+    "Tasas 2008-2012",
+    "Tasas 2013-2016",
+    "Tasas 2017-today"
+  )
+
+  urls <- c(
+   paste0(
+     "https://cdn.bancentral.gov.do/documents/",
+     "estadisticas/sector-monetario-y-financiero/",
+     "documents/tbm_pasiva-1991-2007.xls"),
+
+   paste0(
+     "https://cdn.bancentral.gov.do/documents/",
+     "estadisticas/sector-monetario-y-financiero/",
+     "documents/tbm_pasivad-2008-2012.xls"),
+
+   paste0(
+    "https://cdn.bancentral.gov.do/documents/",
+    "estadisticas/sector-monetario-y-financiero/",
+    "documents/tbm_pasivad-2013-2016.xlsx"),
+
+   paste0(
+     "https://cdn.bancentral.gov.do/documents/",
+     "estadisticas/sector-monetario-y-financiero/",
+     "documents/tbm_pasivad.xlsx")
+   )
+
+  names(urls) <- file_names
+
+  temp_files <- base::sapply(
+    c(".xls", ".xls", ".xlsx", ".xlsx"),
+    \(x) tempfile(fileext = x)
+  ) |>
+    stats::setNames(file_names)
+
+  usethis::ui_info("Downloading files")
+  purrr::walk2(
+    urls, temp_files,
+    utils::download.file, mode = "wb", quiet = TRUE
+  )
+
+  params <- list(
+    list(range = "A157:M266"),
+    list(range = "A14:O88"),
+    list(range = "A11:O67"),
+    list(skip = 10)
+  ) |>
+    purrr::map2(
+      temp_files,
+      \(x, y) append(x, list(path = y, col_names = FALSE))
+    )
+
+  usethis::ui_info("Reading downloaded files")
+  suppressMessages(
+    tasas_list <- purrr::map(
+      seq_along(file_names),
+      \(index) do.call(readxl::read_excel, params[[index]])
+      ) |>
+      stats::setNames(file_names)
+  )
+
+  prepare_data <- function(df, names, start_date) {
+    df |>
+      stats::setNames(names) |>
+      dplyr::filter(stringr::str_detect(mes, "^[A-Z]")) |>
```

Comment:
El archivo de 2017-Today tiene una linea en blanco con "Enero 2022" en la primera celda. Esto hace que cuando se trabaje con este archivo, el resultado tenga un NA. Para corregir esto sugiero modificar esta funcion.

```suggestion
      dplyr::filter(stringr::str_detect(mes, "^[A-Z]") & !is.na(tp_30d)) |>
```